packages/sources/eth-beacon/src/transport/balance.ts

```diff
@@ -0,0 +1,360 @@
+import { ResponseCache } from '@chainlink/external-adapter-framework/cache/response'
+import { Requester } from '@chainlink/external-adapter-framework/util/requester'
+import {
+  AdapterRequest,
+  AdapterResponse,
+  makeLogger,
+  splitArrayIntoChunks,
+} from '@chainlink/external-adapter-framework/util'
+import { BaseEndpointTypes, inputParameters } from '../endpoint/balance'
+import { calculateHttpRequestKey } from '@chainlink/external-adapter-framework/cache'
+import {
+  DEPOSIT_EVENT_LOOKBACK_WINDOW,
+  DEPOSIT_EVENT_TOPIC,
+  formatValueInGwei,
+  parseLittleEndian,
+} from './utils'
+import { ethers } from 'ethers'
+import BigNumber from 'bignumber.js'
+import { DepositEvent_ABI } from '../config/DepositAbi'
+import { PoRBalance } from '@chainlink/external-adapter-framework/adapter/por'
+
+const logger = makeLogger('BalanceTransport')
+
+interface StateResponseSchema {
+  execution_optimistic: false
+  data: ValidatorState[]
+}
+
+interface ValidatorState {
+  index: string
+  balance: string
+  status: string
+  validator: {
+    pubkey: string
+    withdrawal_credentials: string
+    effective_balance: string
+    slashed: boolean
+    activation_eligibility_epoch: string
+    activation_epoch: string
+    exit_epoch: string
+    withdrawable_epoch: string
+  }
+}
+
+export interface BalanceResponse extends PoRBalance {
+  address: string
+}
+
+export type BalanceTransportTypes = BaseEndpointTypes
+
+export type Address = {
+  address: string
+}
+
+export class BalanceTransport implements Transport<BalanceTransportTypes> {
+  name!: string
+  responseCache!: ResponseCache<BalanceTransportTypes>
+  requester!: Requester
+
+  async initialize(
+    dependencies: TransportDependencies<BalanceTransportTypes>,
+    _adapterSettings: BalanceTransportTypes['Settings'],
+    _endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    this.responseCache = dependencies.responseCache
+    this.requester = dependencies.requester
+    this.name = transportName
+  }
+  async foregroundExecute(
+    req: AdapterRequest<typeof inputParameters.validated>,
+    adapterSettings: BalanceTransportTypes['Settings'],
+  ): Promise<AdapterResponse<BalanceTransportTypes['Response']>> {
+    const params = req.requestContext.data
+
+    const url = `/eth/v1/beacon/states/${params.stateId}/validators`
+    const statusList = params.validatorStatus?.join(',')
+    const batchSize = adapterSettings.BATCH_SIZE
+    const responses = []
+
+    // If adapter configured with 0 batch size, put all validators in one request to allow skipping batching
+    if (batchSize === 0) {
+      const addresses = params.addresses.map(({ address }) => address).join(',')
+      responses.push(
+        await this.queryBeaconChain(
+          adapterSettings,
+          req.requestContext.endpointName,
+          url,
+          addresses,
+          statusList,
+        ),
```

Comment:
I think same applies to adapter config as well. It is also used in different places

Reply:
Resolving this comment since it's not necessarily needed for the adapter. With the latest changes it'd only help reduce the clutter in the initialize method